lib/jxl/encode.cc

```diff
@@ -430,6 +457,12 @@ JxlEncoderStatus JxlEncoderAddImageFrame(const JxlEncoderOptions* options,
     c_current = options->enc->metadata.m.color_encoding;
   }
 
+  // needed until API can adjust per-frame animation headers
+  if (options->enc->metadata.m.have_animation) {
+    queued_frame->frame.duration = 10;
+    options->enc->metadata.m.animation.have_timecodes = false;
+  }
+
```

Comment:
Instead of hardcoding all frames to have duration 10, could you propose a new function to set the per-frame info, e.g. a `JxlEncoderSetFrameHeader`?

It's OK if for now it only looks at the duration and is_last (there can be more frame header fields later, like the things I'm proposing to add in https://github.com/libjxl/libjxl/pull/831), but I think that would be a cleaner way to do it than this somewhat ad-hoc solution of a hardcoded duration 10 and deriving is_last from the frame queue being empty or not. If there's an explicit SetFrameHeader function (which would be optional and you would get a default frame header if you don't call it, i.e. is_last=true, duration=0, etc), then you can also encode one frame or a few frames with is_last=false without having to queue _all_ the frames — for long animations, you probably don't want to have to queue all frames simultaneously...